intellij-plugin-verifier/verifier-intellij/src/main/java/com/jetbrains/pluginverifier/usages/internal/kotlin/KtInternalClassUsage.kt

```diff
@@ -0,0 +1,40 @@
+package com.jetbrains.pluginverifier.usages.internal.kotlin
+
+import com.jetbrains.pluginverifier.results.location.ClassLocation
+import com.jetbrains.pluginverifier.results.location.Location
+import com.jetbrains.pluginverifier.results.presentation.ClassGenericsSignatureOption.NO_GENERICS
+import com.jetbrains.pluginverifier.results.presentation.ClassGenericsSignatureOption.WITH_GENERICS
+import com.jetbrains.pluginverifier.results.presentation.ClassOption.FULL_NAME
+import com.jetbrains.pluginverifier.results.presentation.formatClassLocation
+import com.jetbrains.pluginverifier.results.reference.ClassReference
+import com.jetbrains.pluginverifier.usages.formatUsageLocation
+import java.util.*
+
+class KtInternalClassUsage(
+  override val apiReference: ClassReference,
+  override val apiElement: ClassLocation,
+  override val usageLocation: Location
+) : KtInternalModifierUsage() {
+
+  override val problemType: String
+    get() = "Usage of class having Kotlin `internal` visibility modifier."
+
+  override val shortDescription
+    get() = "Internal " + apiElement.elementType.presentableName + " ${apiElement.formatClassLocation(FULL_NAME, NO_GENERICS)} reference"
+
+  override val fullDescription: String
+    get() = buildString {
+      val element = apiElement.elementType.presentableName
+      append("Internal $element ${apiElement.formatClassLocation(FULL_NAME, WITH_GENERICS)}")
+      append(" is referenced in " + usageLocation.formatUsageLocation())
```

Comment:
It looks strange when string concatenations are mixed with templates and explicit `append` calls. I think it would be better to use templates instead of concatenations everywhere, and maybe replace `append` calls by raw string literal.